pkg/controller/mpi_job_controller.go

```diff
@@ -691,6 +692,10 @@ func (c *MPIJobController) getOrCreatePodGroups(mpiJob *kubeflow.MPIJob) (metav1
 		c.recorder.Event(mpiJob, corev1.EventTypeWarning, ErrResourceExists, msg)
 		return nil, fmt.Errorf(msg)
 	}
+
+	if !c.podGroupCtrl.pgSpecsAreEqual(podGroup, newPodGroup) {
```

Comment:
nit:
copy the new spec here. `podGroupCtrl` should just receive the updated object.

Reply:
as discussed, this is fine